uPortal-portlets/src/main/java/org/apereo/portal/portlets/search/google/GoogleCustomSearchService.java

```diff
@@ -64,22 +67,38 @@
                     + CUSTOM_SEARCH_PARAM
                     + "}";
 
-    private String resultSize = "large";
+    private int resultSize = 10;
     private String customSearchId;
+    private String key;
 
     @Value("${org.apereo.portal.portlets.googleWebSearch.search.result.type:googleCustom}")
     private String resultType = "googleCustom";
 
     private RestOperations restOperations;
 
-    public void setResultSize(String resultSize) {
-        this.resultSize = resultSize;
+    public GoogleCustomSearchService() {
+        logger.debug("GoogleCustomSearchService bean instantiated.");
+    }
+
+    public void setResultSize(int resultSize) {
+        // search fails if this parameter is outside the range 1-10, inclusive
+        if (resultSize < 1) {
+            this.resultSize = 1;
+        } else if (resultSize > 10) {
+            this.resultSize = 10;
```

Comment:
If there are > 10 results, can the results outside the first 10 still be accessed?

Reply:
There is the concept of paging in uPortal, and the current page requested is passed in at every call for search results. It might be another issue with the search module.

Reply:
Okay, so should this be managed by `RESULT_SIZE_PARAM` on line 61 then?

Reply:
`resultSize` is the portal configuration value to use as that parameter value. See line 116.